Katana/Middleware/ActionLogger/ActionLogger.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  ActionLogger.swift
+//  Katana
+//
+//  Copyright © 2018 Bending Spoons.
+//  Distributed under the MIT License.
+//  See the LICENSE file for more information.
+//
+
+import Foundation
+
+/**
+ Struct that makes an action logger middleware available
+ */
+public struct ActionLogger {
+  
+  /**
+   This function returns a `StoreMiddleware` that intercepts and logs not black-listed actions
+   
+   - Parameters:
+       - blackList: list of action types that must not be logged
+   - Returns: store middleware that logs not black-listed actions
+   - See: `StoreMiddleware` for details
+   */
+  public static func middleware(blackList: [Action.Type] = []) -> StoreMiddleware {
+  
+    return { getState, dispatch in
+      return { next in
+        return { action in
+          
+          let previousState = getState()
+          next(action)
+          let currentState = getState()
+          
+          let actionType = type(of: action) as Action.Type
```

Comment:
Middleware should do as little as possible in the Katana queue. You should use a dispatch async or something to move things outside this queue as soon as possible